Library/ViewModels/SettingsNewslettersCellViewModel.swift

```diff
@@ -0,0 +1,181 @@
+          .materialize()
+    }
+
+    self.unableToSaveError = updateEvent.errors()
+      .map { env in
+        env.errorMessages.first ?? Strings.profile_settings_error()
+      }
+
+    let previousUserOnError = Signal.merge(initialUser, updatedUser, updateUserAllOn)
+      .combinePrevious()
+      .takeWhen(self.unableToSaveError)
+      .map { previous, _ in previous }
+
+    self.updateCurrentUser = Signal.merge(initialUser, updatedUser, updateUserAllOn, previousUserOnError)
+
+    self.subscribeToAllSwitchIsOn = self.updateCurrentUser
+      .map(userIsSubscribedToAll(user:))
+
+    self.switchIsOn = self.updateCurrentUser
+      .combineLatest(with: newsletter)
+      .map(userIsSubscribed(user:newsletter:))
+
+    // Koala
+    userAttributeChanged
+      .observeValues { attribute, on in
+        switch attribute {
+        case let .newsletter(newsletter):
+          AppEnvironment.current.koala.trackChangeNewsletter(
+            newsletterType: newsletter, sendNewsletter: on, project: nil, context: .settings
+          )
+        default: break
+      }
+    }
+  }
+
+  fileprivate let awakeFromNibProperty = MutableProperty(())
+  public func awakeFromNib() {
+    self.awakeFromNibProperty.value = ()
+  }
+
+  fileprivate let initialUserProperty = MutableProperty<User?>(nil)
+  fileprivate let newsletterProperty = MutableProperty<Newsletter?>(nil)
+  public func configureWith(value: (newsletter: Newsletter, user: User)) {
+    self.newsletterProperty.value = value.newsletter
+    self.initialUserProperty.value = value.user
+  }
+
+  fileprivate let newslettersSwitchTappedProperty = MutableProperty<Bool?>(nil)
+  public func newslettersSwitchTapped(on: Bool) {
+    self.newslettersSwitchTappedProperty.value = on
+  }
+
+  fileprivate let allNewslettersSwitchProperty = MutableProperty<Bool?>(nil)
+  public func allNewslettersSwitchTapped(on: Bool) {
+    self.allNewslettersSwitchProperty.value = on
+  }
+
+  public let showOptInPrompt: Signal<String, NoError>
+  public let subscribeToAllSwitchIsOn: Signal<Bool?, NoError>
+  public let switchIsOn: Signal<Bool?, NoError>
+  public let unableToSaveError: Signal<String, NoError>
+  public let updateCurrentUser: Signal<User, NoError>
+
+  public var inputs: SettingsNewslettersCellViewModelInputs { return self }
+  public var outputs: SettingsNewslettersCellViewModelOutputs { return self }
+}
+
+private func userIsSubscribedToAll(user: User) -> Bool? {
+
+  return user.newsletters.arts == true
+    && user.newsletters.games == true
+    && user.newsletters.happening == true
+    && user.newsletters.invent == true
+    && user.newsletters.promo == true
+    && user.newsletters.weekly == true
+    && user.newsletters.films == true
+    && user.newsletters.publishing == true
+    && user.newsletters.alumni == true
+}
+
+private func userIsSubscribed(user: User, newsletter: Newsletter) -> Bool? {
+  switch newsletter {
```

Comment:
I wonder if maybe instead you could use the `UserAttribute` lens to get this value: `user |> UserAttribute.newsletter(newsletter).lens.view`. Then you don't need to do this switch statement.